### PR TITLE
Add rustcrypto option to use pure-rust crypto

### PR DESCRIPTION
**Pull request by henrikno**
Hi! I was trying to cross-compile my project to arm64, and the only hurdle left was openssl dependency. I managed to move all dependencies away from it from everything except fernet-rs, so I thought I'd add that as an option.
So adding `default-features = false, features = ["rustcrypto"]` should make it compile with musl.

* Added a test in github actions for both with and without the feature flag.
* Also had to switch from chrono to time to get cargo audit check green.

Tested with a simple Dockerfile:
```
FROM rust:latest
RUN rustup target add x86_64-unknown-linux-musl
RUN apt update && apt install -y musl-tools musl-dev
RUN update-ca-certificates
ADD . .
RUN cargo build --release --target x86_64-unknown-linux-musl --no-default-features --features rustcrypto
```

And for cross-compiling
```
FROM rust:latest
RUN apt update && apt install -y musl-tools musl-dev
RUN apt install -y gcc make gcc-aarch64-linux-gnu binutils-aarch64-linux-gnu
RUN rustup target add aarch64-unknown-linux-musl
RUN update-ca-certificates
ADD . .
RUN cargo build --release --target aarch64-unknown-linux-musl --no-default-features --features rustcrypto
```

### CODE REVIEW

File: src/lib.rs
Comment (rozbb) [inline src/lib.rs:302]: Clearer imo: `if !hmac.ct_eq(&expected_hmac).into() {`

Comment (henrikno) [reply src/lib.rs]: Good point. I had to give it a type hint when adding `!`, so it didn't get as clean, but still better than a random u8.

File: src/lib.rs
Comment (henrikno) [reply src/lib.rs]: Fixed

File: Cargo.toml
Comment (henrikno) [reply Cargo.toml:17]: Added https://github.com/mozilla-services/fernet-rs/pull/65/files#diff-b1a35a68f14e696205874893c07fd24fdb88882b47c23cc0e0c80a30c7d53759R33

File: src/lib.rs
Comment (henrikno) [reply src/lib.rs]: Made this into a struct with is a bit easier to document. I attempted to avoid using owned fields (specifically for ciphertext), but couldn't get the lifetimes to work out (there's a lot of overlapping references), but at least the secrets I've seen are fairly small so it shouldn't be _too_ bad.

File: src/lib.rs
Comment (rozbb) [inline src/lib.rs:35]: This is technically fine, but I think `openssl` should be referenced rather than `default`. This is also what the dalek library does

**Comment by rozbb** [comment]
Changes look good! Left 1 comment. Also: this does not compile right now: `cargo test --no-default-features --features "rustcrypto"` fails.

The fix is easy though: add `features = ["alloc"]` to the `cbc` dependency.

File: src/lib.rs
Comment (henrikno) [reply src/lib.rs]: Ah, for some reason I thought it wouldn't work since openssl isn't a "feature", but a dependency, but dependencies are implicitly a feature. Fixed.

**Comment by henrikno** [comment]
Re: `alloc`, added. Strange that it somehow didn't need it on macOS, but needs it it on Linux.

**Comment by rozbb** [comment]
Looks great! Someone with merge permissions will hopefully chime in soon. Thanks again for the PR

**Comment by eipipuz** [comment]
Eagerly awaiting for this merge. This looks like it will allow me to cross-compile from mac to windows 👯 🎉

**Comment by GnomedDev** [comment]
Also waiting for this merge, would allow me to get rid of openssl from one of my projects

**Review by jrconlin — APPROVED** [review]

**Comment by jrconlin** [comment]
Sorry for the delay on this. Got lost in the shuffle. Feel free to reach out directly in the future.

**Comment by rozbb** [comment]
Will do, and no worries. Thank you!

**Pull request merged**
